core/server/api_container/server/startosis_engine/startosis_packages/git_package_content_provider/git_package_content_provider.go

```diff
@@ -366,6 +379,50 @@ func (provider *GitPackageContentProvider) atomicClone(parsedURL *ParsedGitURL)
 	return nil
 }
 
+func replaceAbsoluteLocator(absoluteLocator string, packageReplaceOptions map[string]string) string {
+	if absoluteLocator == "" {
+		return absoluteLocator
+	}
+
+	found, packageToBeReplaced, replaceWithPackage := findPackageReplace(absoluteLocator, packageReplaceOptions)
+	if found {
+		// we skip if it's a local replace because we will use the same absolute locator
+		// due the file was already uploaded in the enclave's package cache
+		if isLocalDependencyReplace(replaceWithPackage) {
+			return absoluteLocator
+		}
+		replacedAbsoluteLocator := strings.Replace(absoluteLocator, packageToBeReplaced, replaceWithPackage, onlyOneReplace)
+		logrus.Debugf("absoluteLocator '%s' replaced with '%s'", absoluteLocator, replacedAbsoluteLocator)
+		return replacedAbsoluteLocator
+	}
+
+	return absoluteLocator
+}
+
+func findPackageReplace(absoluteLocator string, packageReplaceOptions map[string]string) (bool, string, string) {
+	pathToAnalyze := absoluteLocator
+	for {
+		numberSlashes := strings.Count(pathToAnalyze, urlPathSeparator)
```

Comment:
can we just parseGitUrl this or something? and replace the `relativeRepoFile` or something. We have a lot of extra string parsing going on